arraykit.c

```diff
@@ -288,6 +289,22 @@ ArrayGO_extend(ArrayGOObject *self, PyObject *values)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+ArrayGO_getnewargs(ArrayGOObject *self, PyObject *Py_UNUSED(unused))
+{
+    if (self->list && update_array_cache(self)) {
+        return NULL;
+    }
+    PyObject *args = PyTuple_New(1);
+    if (!args) {
+        return NULL;
+    }
+    PyTuple_SET_ITEM(args, 0, self->array);
```

Comment:
This is what's causing your segfault: you're inserting an item into your tuple without increasing the item's reference count. `PyTuple_SET_ITEM` doesn't do this for you (it "steals" a reference to `self->array`).

```suggestion
    Py_INCREF(self->array);
    PyTuple_SET_ITEM(args, 0, self->array);
```

Reply:
Thanks for that explanation. With this change (and the other), however, we are still segfaulting!

Reply:
I also tried using `PyArray_INCREF((PyArrayObject *)self->array);` instead of `Py_INCREF`, but that did not stop the segfault.

Reply:
Yeah, see my comment below. I don't think the segfault actually happens here... it happens when you access `values` after unpickling!

I'm going to make a patch to move all of the logic in `__init__` to `__new__`. That should fix this.